<a href="https://www.kaggle.com/code/mahmoudashrey/google-play-store?scriptVersionId=348819184" target="_blank"><img align="left" alt="Kaggle" title="Open in Kaggle" src="https://kaggle.com/static/images/open-in-kaggle.svg"></a>

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px
import plotly.graph_objects as go

In [2]:
sns.set_theme(style='whitegrid')

In [3]:
df = pd.read_csv("/kaggle/input/datasets/lava18/google-play-store-apps/googleplaystore.csv")

In [4]:
df.head()

,App,Category,Rating,Reviews,Size,Installs,Type,Price,Content Rating,Genres,Last Updated,Current Ver,Android Ver
0,Photo Editor & Candy Camera & Grid & ScrapBook,ART_AND_DESIGN,4.1,159,19M,"10,000+",Free,0,Everyone,Art & Design,"January 7, 2018",1.0.0,4.0.3 and up
1,Coloring book moana,ART_AND_DESIGN,3.9,967,14M,"500,000+",Free,0,Everyone,Art & Design;Pretend Play,"January 15, 2018",2.0.0,4.0.3 and up
2,"U Launcher Lite – FREE Live Cool Themes, Hide ...",ART_AND_DESIGN,4.7,87510,8.7M,"5,000,000+",Free,0,Everyone,Art & Design,"August 1, 2018",1.2.4,4.0.3 and up
3,Sketch - Draw & Paint,ART_AND_DESIGN,4.5,215644,25M,"50,000,000+",Free,0,Teen,Art & Design,"June 8, 2018",Varies with device,4.2 and up
4,Pixel Draw - Number Art Coloring Book,ART_AND_DESIGN,4.3,967,2.8M,"100,000+",Free,0,Everyone,Art & Design;Creativity,"June 20, 2018",1.1,4.4 and up


In [5]:
df.shape

(10841, 13)

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10841 non-null  object 
 1   Category        10841 non-null  object 
 2   Rating          9367 non-null   float64
 3   Reviews         10841 non-null  object 
 4   Size            10841 non-null  object 
 5   Installs        10841 non-null  object 
 6   Type            10840 non-null  object 
 7   Price           10841 non-null  object 
 8   Content Rating  10840 non-null  object 
 9   Genres          10841 non-null  object 
 10  Last Updated    10841 non-null  object 
 11  Current Ver     10833 non-null  object 
 12  Android Ver     10838 non-null  object 
dtypes: float64(1), object(12)
memory usage: 1.1+ MB


In [7]:
df.describe()

,Rating
count,9367.000000
mean,4.193338
std,0.537431
min,1.000000
25%,4.000000
50%,4.300000
75%,4.500000
max,19.000000


In [8]:
print(df.isnull().sum())

print("-" * 40)

print((df.isnull().sum() / len(df) * 100).round(2))

App                  0
Category             0
Rating            1474
Reviews              0
Size                 0
Installs             0
Type                 1
Price                0
Content Rating       1
Genres               0
Last Updated         0
Current Ver          8
Android Ver          3
dtype: int64
----------------------------------------
App                0.00
Category           0.00
Rating            13.60
Reviews            0.00
Size               0.00
Installs           0.00
Type               0.01
Price              0.00
Content Rating     0.01
Genres             0.00
Last Updated       0.00
Current Ver        0.07
Android Ver        0.03
dtype: float64


In [9]:
df.duplicated().sum()

np.int64(483)

In [10]:
df['App'].nunique()

9660

In [11]:
df['App'].value_counts().head(10)

App
ROBLOX                                               9
CBS Sports App - Scores, News, Stats & Watch Live    8
8 Ball Pool                                          7
Candy Crush Saga                                     7
ESPN                                                 7
Duolingo: Learn Languages Free                       7
Temple Run 2                                         6
Bowmasters                                           6
Helix Jump                                           6
Nick                                                 6
Name: count, dtype: int64

In [12]:
df[df['Rating'] > 5][['App', 'Category', 'Rating', 'Reviews', 'Installs', 'Type']]

,App,Category,Rating,Reviews,Installs,Type
10472,Life Made WI-Fi Touchscreen Photo Frame,1.9,19.0,3.0M,Free,0


In [13]:
df[df['App'] == 'ROBLOX']

,App,Category,Rating,Reviews,Size,Installs,Type,Price,Content Rating,Genres,Last Updated,Current Ver,Android Ver
1653,ROBLOX,GAME,4.5,4447388,67M,"100,000,000+",Free,0,Everyone 10+,Adventure;Action & Adventure,"July 31, 2018",2.347.225742,4.1 and up
1701,ROBLOX,GAME,4.5,4447346,67M,"100,000,000+",Free,0,Everyone 10+,Adventure;Action & Adventure,"July 31, 2018",2.347.225742,4.1 and up
1748,ROBLOX,GAME,4.5,4448791,67M,"100,000,000+",Free,0,Everyone 10+,Adventure;Action & Adventure,"July 31, 2018",2.347.225742,4.1 and up
1841,ROBLOX,GAME,4.5,4449882,67M,"100,000,000+",Free,0,Everyone 10+,Adventure;Action & Adventure,"July 31, 2018",2.347.225742,4.1 and up
1870,ROBLOX,GAME,4.5,4449910,67M,"100,000,000+",Free,0,Everyone 10+,Adventure;Action & Adventure,"July 31, 2018",2.347.225742,4.1 and up
2016,ROBLOX,FAMILY,4.5,4449910,67M,"100,000,000+",Free,0,Everyone 10+,Adventure;Action & Adventure,"July 31, 2018",2.347.225742,4.1 and up
2088,ROBLOX,FAMILY,4.5,4450855,67M,"100,000,000+",Free,0,Everyone 10+,Adventure;Action & Adventure,"July 31, 2018",2.347.225742,4.1 and up
2206,ROBLOX,FAMILY,4.5,4450890,67M,"100,000,000+",Free,0,Everyone 10+,Adventure;Action & Adventure,"July 31, 2018",2.347.225742,4.1 and up
4527,ROBLOX,FAMILY,4.5,4443407,67M,"100,000,000+",Free,0,Everyone 10+,Adventure;Action & Adventure,"July 31, 2018",2.347.225742,4.1 and up


In [14]:
df = df.drop_duplicates()

df.shape

(10358, 13)

In [15]:
df[df['Rating'] > 5]

,App,Category,Rating,Reviews,Size,Installs,Type,Price,Content Rating,Genres,Last Updated,Current Ver,Android Ver
10472,Life Made WI-Fi Touchscreen Photo Frame,1.9,19.0,3.0M,"1,000+",Free,0,Everyone,NaN,"February 11, 2018",1.0.19,4.0 and up,NaN


In [16]:
df = df[df['Rating'].isna() | (df['Rating'] <= 5)].copy()

df.shape

(10357, 13)

In [17]:
df['Rating'].max()

5.0

In [18]:
df[~df['Reviews'].str.replace(',', '', regex = False).str.isnumeric()]['Reviews'].unique()

array([], dtype=object)

In [19]:
df['Reviews'] = df['Reviews'].str.replace(',', '', regex = False).astype(int)

In [20]:
df['Reviews'].dtype

dtype('int64')

In [21]:
df['Reviews'].describe()

count    1.035700e+04
mean     4.059046e+05
std      2.696778e+06
min      0.000000e+00
25%      3.200000e+01
50%      1.680000e+03
75%      4.641600e+04
max      7.815831e+07
Name: Reviews, dtype: float64

In [22]:
df['Installs'].unique()

array(['10,000+', '500,000+', '5,000,000+', '50,000,000+', '100,000+',
       '50,000+', '1,000,000+', '10,000,000+', '5,000+', '100,000,000+',
       '1,000,000,000+', '1,000+', '500,000,000+', '50+', '100+', '500+',
       '10+', '1+', '5+', '0+', '0'], dtype=object)

In [23]:
df['Installs'] = (
    df['Installs']
    .str.replace(',', '', regex = False)
    .str.replace('+', '', regex = False)
    .astype(int)
)

In [24]:
df['Installs'].dtype

dtype('int64')

In [25]:
df['Installs'].describe()

count    1.035700e+04
mean     1.415776e+07
std      8.023955e+07
min      0.000000e+00
25%      1.000000e+03
50%      1.000000e+05
75%      1.000000e+06
max      1.000000e+09
Name: Installs, dtype: float64

In [26]:
df['Price'].unique()

array(['0', '$4.99', '$3.99', '$6.99', '$1.49', '$2.99', '$7.99', '$5.99',
       '$3.49', '$1.99', '$9.99', '$7.49', '$0.99', '$9.00', '$5.49',
       '$10.00', '$24.99', '$11.99', '$79.99', '$16.99', '$14.99',
       '$1.00', '$29.99', '$12.99', '$2.49', '$10.99', '$1.50', '$19.99',
       '$15.99', '$33.99', '$74.99', '$39.99', '$3.95', '$4.49', '$1.70',
       '$8.99', '$2.00', '$3.88', '$25.99', '$399.99', '$17.99',
       '$400.00', '$3.02', '$1.76', '$4.84', '$4.77', '$1.61', '$2.50',
       '$1.59', '$6.49', '$1.29', '$5.00', '$13.99', '$299.99', '$379.99',
       '$37.99', '$18.99', '$389.99', '$19.90', '$8.49', '$1.75',
       '$14.00', '$4.85', '$46.99', '$109.99', '$154.99', '$3.08',
       '$2.59', '$4.80', '$1.96', '$19.40', '$3.90', '$4.59', '$15.46',
       '$3.04', '$4.29', '$2.60', '$3.28', '$4.60', '$28.99', '$2.95',
       '$2.90', '$1.97', '$200.00', '$89.99', '$2.56', '$30.99', '$3.61',
       '$394.99', '$1.26', '$1.20', '$1.04'], dtype=object)

In [27]:
df['Price'] = df['Price'].str.replace('$', '', regex = False).astype(float)

In [28]:
df['Price'].dtype

dtype('float64')

In [29]:
df['Price'].describe()

count    10357.000000
mean         1.030800
std         16.278625
min          0.000000
25%          0.000000
50%          0.000000
75%          0.000000
max        400.000000
Name: Price, dtype: float64

In [30]:
df['Size'].unique()

array(['19M', '14M', '8.7M', '25M', '2.8M', '5.6M', '29M', '33M', '3.1M',
       '28M', '12M', '20M', '21M', '37M', '2.7M', '5.5M', '17M', '39M',
       '31M', '4.2M', '7.0M', '23M', '6.0M', '6.1M', '4.6M', '9.2M',
       '5.2M', '11M', '24M', 'Varies with device', '9.4M', '15M', '10M',
       '1.2M', '26M', '8.0M', '7.9M', '56M', '57M', '35M', '54M', '201k',
       '3.6M', '5.7M', '8.6M', '2.4M', '27M', '2.5M', '16M', '3.4M',
       '8.9M', '3.9M', '2.9M', '38M', '32M', '5.4M', '18M', '1.1M',
       '2.2M', '4.5M', '9.8M', '52M', '9.0M', '6.7M', '30M', '2.6M',
       '7.1M', '3.7M', '22M', '7.4M', '6.4M', '3.2M', '8.2M', '9.9M',
       '4.9M', '9.5M', '5.0M', '5.9M', '13M', '73M', '6.8M', '3.5M',
       '4.0M', '2.3M', '7.2M', '2.1M', '42M', '7.3M', '9.1M', '55M',
       '23k', '6.5M', '1.5M', '7.5M', '51M', '41M', '48M', '8.5M', '46M',
       '8.3M', '4.3M', '4.7M', '3.3M', '40M', '7.8M', '8.8M', '6.6M',
       '5.1M', '61M', '66M', '79k', '8.4M', '118k', '44M', '695k', '1.6M',
     

In [31]:
def convert_size(size):
    if size.endswith('M'):
        return float(size[:-1])
    elif size.endswith('k'):
        return float(size[:-1]) / 1024
    else:
        return np.nan

df['Size'] = df['Size'].apply(convert_size)

In [32]:
df['Size'].dtype

dtype('float64')

In [33]:
df['Size'].describe()

count    8831.000000
mean       21.287413
std        22.540591
min         0.008301
25%         4.700000
50%        13.000000
75%        29.000000
max       100.000000
Name: Size, dtype: float64

In [34]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 10357 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10357 non-null  object 
 1   Category        10357 non-null  object 
 2   Rating          8892 non-null   float64
 3   Reviews         10357 non-null  int64  
 4   Size            8831 non-null   float64
 5   Installs        10357 non-null  int64  
 6   Type            10356 non-null  object 
 7   Price           10357 non-null  float64
 8   Content Rating  10357 non-null  object 
 9   Genres          10357 non-null  object 
 10  Last Updated    10357 non-null  object 
 11  Current Ver     10349 non-null  object 
 12  Android Ver     10355 non-null  object 
dtypes: float64(3), int64(2), object(8)
memory usage: 1.1+ MB


In [35]:
df.isnull().sum()

App                  0
Category             0
Rating            1465
Reviews              0
Size              1526
Installs             0
Type                 1
Price                0
Content Rating       0
Genres               0
Last Updated         0
Current Ver          8
Android Ver          2
dtype: int64

In [36]:
df[df['Type'].isna() | df['Current Ver'].isna() | df['Android Ver'].isna()]

,App,Category,Rating,Reviews,Size,Installs,Type,Price,Content Rating,Genres,Last Updated,Current Ver,Android Ver
15,Learn To Draw Kawaii Characters,ART_AND_DESIGN,3.2,55,2.700000,5000,Free,0.00,Everyone,Art & Design,"June 6, 2018",NaN,4.2 and up
1553,Market Update Helper,LIBRARIES_AND_DEMO,4.1,20145,0.010742,1000000,Free,0.00,Everyone,Libraries & Demo,"February 12, 2013",NaN,1.5 and up
4453,[substratum] Vacuum: P,PERSONALIZATION,4.4,230,11.000000,1000,Paid,1.49,Everyone,Personalization,"July 20, 2018",4.4,NaN
4490,Pi Dark [substratum],PERSONALIZATION,4.5,189,2.100000,10000,Free,0.00,Everyone,Personalization,"March 27, 2018",1.1,NaN
6322,Virtual DJ Sound Mixer,TOOLS,4.2,4010,8.700000,500000,Free,0.00,Everyone,Tools,"May 10, 2017",NaN,4.0 and up
6803,BT Master,FAMILY,NaN,0,0.216797,100,Free,0.00,Everyone,Education,"November 6, 2016",NaN,1.6 and up
7333,Dots puzzle,FAMILY,4.0,179,14.000000,50000,Paid,0.99,Everyone,Puzzle,"April 18, 2018",NaN,4.0 and up
7407,Calculate My IQ,FAMILY,NaN,44,7.200000,10000,Free,0.00,Everyone,Entertainment,"April 3, 2017",NaN,2.3 and up
7730,UFO-CQ,TOOLS,NaN,1,0.231445,10,Paid,0.99,Everyone,Tools,"July 4, 2016",NaN,2.0 and up
9148,Command & Conquer: Rivals,FAMILY,NaN,0,NaN,0,NaN,0.00,Everyone 10+,Strategy,"June 28, 2018",Varies with device,Varies with device


In [37]:
df.loc[df['Type'].isna(), 'Type'] = 'Free'

In [38]:
df['Current Ver'] = df['Current Ver'].fillna('Unknown')
df['Android Ver'] = df['Android Ver'].fillna('Unknown')

In [39]:
df[['Type', 'Current Ver', 'Android Ver']].isnull().sum()

Type           0
Current Ver    0
Android Ver    0
dtype: int64

In [40]:
df['Last Updated'] = pd.to_datetime(df['Last Updated'])

In [41]:
df['Last Updated'].dtype

dtype('<M8[ns]')

In [42]:
df.isnull().sum()

App                  0
Category             0
Rating            1465
Reviews              0
Size              1526
Installs             0
Type                 0
Price                0
Content Rating       0
Genres               0
Last Updated         0
Current Ver          0
Android Ver          0
dtype: int64

In [43]:
df.dtypes

App                       object
Category                  object
Rating                   float64
Reviews                    int64
Size                     float64
Installs                   int64
Type                      object
Price                    float64
Content Rating            object
Genres                    object
Last Updated      datetime64[ns]
Current Ver               object
Android Ver               object
dtype: object

In [44]:
df['Rating'].describe()

count    8892.000000
mean        4.187877
std         0.522377
min         1.000000
25%         4.000000
50%         4.300000
75%         4.500000
max         5.000000
Name: Rating, dtype: float64

In [45]:
df.shape

(10357, 13)

## App Distribution by Category

Which app categories have the largest number of applications in the Google Play Store dataset?

In [46]:
category_counts = df['Category'].value_counts().reset_index()
category_counts.columns = ['Category', 'App Count']

fig = px.bar(
    category_counts,
    x = 'App Count',
    y = 'Category',
    orientation = 'h',
    title = 'Number of Apps by Category',
    labels = {
        'App Count': 'Number of Apps',
        'Category': 'Category'
    }
)

fig.update_layout(yaxis = {'categoryorder': 'total ascending'})

fig.show()

### Insight

The FAMILY category has the highest number of apps by a large margin, followed by GAME and TOOLS. Most other categories contain considerably fewer apps, indicating that the app market in this dataset is heavily concentrated in a few major categories.

## Free vs Paid Apps

What is the distribution of free and paid applications in the Google Play Store dataset?

In [47]:
type_counts = df['Type'].value_counts().reset_index()
type_counts.columns = ['Type', 'App Count']

fig = px.pie(
    type_counts,
    names = 'Type',
    values = 'App Count',
    hole = 0.5,
    title = 'Free vs Paid Apps',
)

fig.show()

### Insight

Free apps dominate the Google Play Store dataset, accounting for approximately 92.6% of all apps, while paid apps represent only 7.4%. This indicates that the free-to-download model is far more common than the paid model in this dataset.

## App Rating Distribution

How are app ratings distributed across applications, and what rating range is most common?

In [48]:
fig = px.histogram(
    df.dropna(subset = ['Rating']),
    x = 'Rating',
    nbins = 20,
    title = 'Distribution of App Ratings',
    labels = {'Rating': 'Rating'}
)

fig.show()

### Insight

App ratings are heavily concentrated at the higher end of the scale, with most applications receiving ratings between 4.0 and 4.7. Ratings below 3.0 are relatively uncommon, indicating that highly rated apps dominate the dataset.

## Reviews vs Installs

Is there a relationship between the number of installs and the number of user reviews?

In [49]:
fig = px.scatter(
    df,
    x = 'Installs',
    y = 'Reviews',
    title = 'Reviews vs Installs',
    labels = {
        'Installs': 'Number of Installs',
        'Reviews': 'Number of Reviews'
    },
    log_x = True,
    log_y = True,
    hover_data = ['App', 'Category']
)

fig.show()

### Insight

There is a strong positive relationship between the number of installs and user reviews. Apps with higher install counts generally receive substantially more reviews, although the number of reviews varies considerably among apps with similar install levels.

## Most Installed Apps

Which applications have the highest number of installs in the Google Play Store dataset?

In [50]:
top_apps = (
    df.groupby('App', as_index = False)['Installs']
    .max()
    .sort_values('Installs', ascending = False)
    .head(10)
)

fig = px.bar(
    top_apps.sort_values('Installs'),
    x = 'Installs',
    y = 'App',
    orientation = 'h',
    title = 'Top 10 Most Installed Apps',
    labels = {
        'Installs': 'Number of Installs',
        'App': 'App'
    }
)

fig.show()

### Insight

The top 10 most installed apps all appear in the 1 billion+ install category, showing that these apps have reached the highest install tier recorded in the dataset. This also highlights a limitation of the `Installs` variable, as the values are grouped into install ranges rather than exact download counts.

## Ratings by App Type

Do paid apps receive higher ratings than free apps?

In [51]:
fig = px.box(
    df.dropna(subset = ['Rating']),
    x = 'Type',
    y = 'Rating',
    title = 'Ratings by App Type',
    labels = {
        'Type': 'App Type',
        'Rating': 'Rating'
    }
)

fig.show()

### Insight

Paid apps have a slightly higher median rating than free apps, with the median rating around 4.4 compared with approximately 4.3 for free apps. However, both groups show a similar overall distribution and contain lower-rated outliers, suggesting that app type alone does not strongly determine user ratings.

In [52]:
reviews_df = pd.read_csv("/kaggle/input/datasets/lava18/google-play-store-apps/googleplaystore_user_reviews.csv")

In [53]:
reviews_df.head()

,App,Translated_Review,Sentiment,Sentiment_Polarity,Sentiment_Subjectivity
0,10 Best Foods for You,I like eat delicious food. That's I'm cooking ...,Positive,1.00,0.533333
1,10 Best Foods for You,This help eating healthy exercise regular basis,Positive,0.25,0.288462
2,10 Best Foods for You,NaN,NaN,NaN,NaN
3,10 Best Foods for You,Works great especially going grocery store,Positive,0.40,0.875000
4,10 Best Foods for You,Best idea us,Positive,1.00,0.300000


In [54]:
reviews_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 64295 entries, 0 to 64294
Data columns (total 5 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   App                     64295 non-null  object 
 1   Translated_Review       37427 non-null  object 
 2   Sentiment               37432 non-null  object 
 3   Sentiment_Polarity      37432 non-null  float64
 4   Sentiment_Subjectivity  37432 non-null  float64
dtypes: float64(2), object(3)
memory usage: 2.5+ MB


In [55]:
reviews_df.duplicated().sum()

np.int64(33616)

In [56]:
reviews_df.isnull().sum()

App                           0
Translated_Review         26868
Sentiment                 26863
Sentiment_Polarity        26863
Sentiment_Subjectivity    26863
dtype: int64

In [57]:
reviews_df['Sentiment'].value_counts(dropna = False)

Sentiment
NaN         26863
Positive    23998
Negative     8271
Neutral      5163
Name: count, dtype: int64

In [58]:
reviews_df = reviews_df.drop_duplicates()

reviews_df.shape

(30679, 5)

In [59]:
sentiment_df = reviews_df.dropna(subset=['Sentiment']).copy()

sentiment_df.shape

(29697, 5)

In [60]:
sentiment_df['Sentiment'].value_counts()

Sentiment
Positive    19015
Negative     6321
Neutral      4361
Name: count, dtype: int64

## User Sentiment Distribution

What is the overall sentiment of users toward the apps based on their reviews?

In [61]:
sentiment_counts = sentiment_df['Sentiment'].value_counts().reset_index()
sentiment_counts.columns = ['Sentiment', 'Review Count']

fig = px.pie(
    sentiment_counts,
    names = 'Sentiment',
    values = 'Review Count',
    hole = 0.5,
    title = 'User Sentiment Distribution'
)

fig.show()

### Insight

Positive reviews account for 64.0% of the analyzed reviews, making them the dominant sentiment. Negative reviews represent 21.3%, while neutral reviews account for 14.7%, indicating that user feedback is generally more positive than negative in this dataset.

## App Rating vs User Sentiment

Do apps with higher Google Play ratings tend to receive more positive user reviews?

In [62]:
sentiment_summary = (
    sentiment_df
    .groupby('App')['Sentiment']
    .value_counts(normalize = True)
    .unstack(fill_value = 0)
    .reset_index()
)

sentiment_summary.head()

Sentiment,App,Negative,Neutral,Positive
0,10 Best Foods for You,0.052632,0.115789,0.831579
1,104 找工作 - 找工作 找打工 找兼職 履歷健檢 履歷診療室,0.027778,0.222222,0.750000
2,11st,0.175000,0.250000,0.575000
3,1800 Contacts - Lens Store,0.075000,0.125000,0.800000
4,1LINE – One Line with One Touch,0.210526,0.078947,0.710526


In [63]:
app_analysis = df[['App', 'Rating']].dropna(subset=['Rating']).drop_duplicates('App')

app_analysis = app_analysis.merge(
    sentiment_summary,
    on='App',
    how='inner'
)

app_analysis.head()

,App,Rating,Negative,Neutral,Positive
0,Coloring book moana,3.9,0.350000,0.100000,0.550000
1,Garden Coloring Book,4.4,0.111111,0.177778,0.711111
2,FlipaClip - Cartoon animation,4.3,0.000000,0.000000,1.000000
3,Boys Photo Editor - Six Pack & Men's Suit,4.1,0.138889,0.277778,0.583333
4,Colorfit - Drawing & Coloring,4.7,0.210526,0.026316,0.763158


In [64]:
fig = px.scatter(
    app_analysis,
    x = 'Rating',
    y = 'Positive',
    title = 'App Rating vs Positive Review Sentiment',
    labels = {
        'Rating': 'App Rating',
        'Positive': 'Positive Review Percentage'
    },
    hover_data = ['App']
)

fig.update_yaxes(tickformat = '.0%')

fig.show()

### Insight

Apps with higher ratings generally tend to have a higher percentage of positive reviews. However, the wide spread of points at most rating levels shows that app ratings do not always align closely with review sentiment, suggesting that other factors may influence users' written feedback.

## User Sentiment by App Category

Do user sentiments differ across different app categories?

In [65]:
# Merge reviews with app categories
reviews_with_category = sentiment_df.merge(
    df[['App', 'Category']],
    on = 'App',
    how = 'inner'
)

# Select top 10 categories by number of reviews
top_categories = (
    reviews_with_category['Category']
    .value_counts()
    .head(10)
    .index
)

category_sentiment = (
    reviews_with_category[
        reviews_with_category['Category'].isin(top_categories)
    ]
    .groupby('Category')['Sentiment']
    .value_counts(normalize = True)
    .unstack(fill_value = 0)
    .reset_index()
)

category_sentiment

Sentiment,Category,Negative,Neutral,Positive
0,DATING,0.216543,0.180229,0.603228
1,FAMILY,0.277183,0.088732,0.634085
2,FINANCE,0.221379,0.138520,0.640101
3,GAME,0.375013,0.044550,0.580437
4,HEALTH_AND_FITNESS,0.113696,0.116341,0.769963
5,PHOTOGRAPHY,0.199477,0.125572,0.674951
6,PRODUCTIVITY,0.186707,0.144411,0.668882
7,SPORTS,0.244099,0.131056,0.624845
8,TOOLS,0.178099,0.212395,0.609507
9,TRAVEL_AND_LOCAL,0.212465,0.168272,0.619263


In [66]:
fig = px.bar(
    category_sentiment,
    x = 'Category',
    y = ['Positive', 'Negative', 'Neutral'],
    title = 'User Sentiment by App Category',
    labels = {
        'value': 'Percentage of Reviews',
        'Category': 'App Category',
        'variable': 'Sentiment'
    },
    barmode = 'group'
)

fig.update_yaxes(tickformat = '.0%')

fig.show()

### Insight

Positive sentiment dominates across all of the top 10 app categories. Health & Fitness has the highest share of positive reviews at approximately 77%, while Game has the highest share of negative reviews at about 38%. This suggests that user satisfaction varies noticeably across categories, with some categories receiving more favorable feedback than others.

# Key Findings

- The FAMILY category has the largest number of apps in the dataset, followed by GAME and TOOLS, showing that app distribution is concentrated in a few major categories.

- Free applications dominate the dataset, representing approximately 92.6% of all apps, while paid applications account for only 7.4%.

- App ratings are generally high, with most ratings concentrated between 4.0 and 4.7.

- There is a strong positive relationship between installs and reviews. Apps with more installs generally receive more user reviews.

- The most installed apps are concentrated in the 1 billion+ install tier, highlighting the high popularity of major apps in the dataset.

- Paid apps have a slightly higher median rating than free apps, but the difference is relatively small, suggesting that pricing alone does not strongly determine user ratings.

- Positive sentiment is dominant among user reviews, accounting for approximately 64% of analyzed reviews.

- Higher app ratings generally tend to be associated with a higher percentage of positive reviews, although the relationship is not perfect.

- Positive sentiment dominates across the major app categories, while the proportion of negative reviews varies noticeably between categories.

# Conclusion

This analysis explored Google Play Store applications from multiple perspectives, including app categories, pricing, ratings, installs, reviews, and user sentiment. The results show that the dataset is dominated by free applications and highly rated apps, while installs and reviews have a strong positive relationship. User feedback is also predominantly positive, although sentiment varies across app categories. Overall, combining app-level metrics with user review sentiment provides a more complete view of app popularity and user satisfaction.